src/phone-number-input/phone-number-input.tsx

```diff
@@ -25,26 +23,29 @@ export const PhoneNumberInput = ({
     // =============================================================================
     // CONST, STATE, REF
     // =============================================================================
-    const [inputNumber, setInputNumber] = useState(value?.number || "");
-    const [selectedCountryCode, setSelectedCountryCode] = useState(
-        value?.countryCode || ""
-    );
+    const options = useMemo(() => PhoneNumberInputHelper.getCountries, []);
 
-    const selectedOption: CountryValue =
-        PhoneNumberInputHelper.getCountries.filter(
-            (country: CountryValue) =>
-                country.countryCode === selectedCountryCode.replace("+", "")
-        )[0];
+    const [selectedCountry, setSelectedCountry] = useState<
+        CountryValue | undefined
+    >(undefined);
+    const [inputValue, setInputValue] = useState<string>("");
 
     const nodeRef = useRef();
 
     // =============================================================================
     // EFFECTS
     // =============================================================================
     useEffect(() => {
-        const formattedNumber = getFormattedNumber(value?.number);
-        setInputNumber(formattedNumber);
-    }, [value]);
+        const selectedOption = options.filter(
+            (country) =>
+                country.countryCode === normaliseCountryCode(value?.countryCode)
+        )[0];
+
+        setSelectedCountry(selectedOption);
+        setInputValue(
+            PhoneNumberInputHelper.formatNumber(value?.number, selectedOption)
+        );
+    }, [options, value]);
```

Comment:
continuing from above, no need to use options as a hook dependency

also compare the actual values instead of the object ref to prevent useEffect from running unnecessarily

```suggestion
    }, [value.?countryCode, value?.number]);
```